deployment/chart/templates/deployment.yaml

```diff
@@ -0,0 +1,113 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "code-server.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "code-server.name" . }}
+    helm.sh/chart: {{ include "code-server.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  replicas: 1
+  strategy:
+    type: Recreate
```

Comment:
Are you sure Recreate is the best strategy for this? I prefer a rolling update since it allows fallback when the container fails to deploy. Case and point, practice A/B

Reply:
Yes i agree with the theory, but when using a ReadWriteOnce volume more than one pod cannot mount the volume, that will cause the deployment to fail.
Another strategy would be to set `MaxSurge=0 and MaxUnavailable=1.` to the RollingUpdate strategy but it has the same effect of Recreate.

Reply:
I don't think most dev/test K8s clusters has restrictions with PVCs, you might as well do ReadWriteMany if that's the case. If we find a lot of issues with this then we can revert back to RWO

Reply:
ReadWriteMany cannot be used for most of the cloud vendor disks, including AWS, GCP and Azure, unless you use a network file-system, this is the safest and most compatible choice.

Reply:
As you can see here: https://github.com/helm/charts/search?q=ReadWriteOnce&unscoped_q=ReadWriteOnce it's the common choice rather then https://github.com/helm/charts/search?q=ReadWriteMany&unscoped_q=ReadWriteMany which points directly to NFS-like charts.